lib/widgets/action_sheet.dart

```diff
@@ -143,6 +144,225 @@ class ActionSheetCancelButton extends StatelessWidget {
   }
 }
 
+/// Show a sheet of actions you can take on a topic.
+void showTopicActionSheet(BuildContext context, {
+  required int channelId,
+  required String topic,
+}) {
+  final narrow = TopicNarrow(channelId, topic);
+  UserTopicUpdateButton button({
+    UserTopicVisibilityPolicy? from,
+    required UserTopicVisibilityPolicy to,
+  }) {
+    return UserTopicUpdateButton(
+      currentVisibilityPolicy: from,
+      newVisibilityPolicy: to,
+      narrow: narrow,
+      pageContext: context);
+  }
+
+  final mute =                 button(to:   UserTopicVisibilityPolicy.muted);
+  final unmute =               button(from: UserTopicVisibilityPolicy.muted,
+                                      to:   UserTopicVisibilityPolicy.none);
+  final unmuteInMutedChannel = button(to:   UserTopicVisibilityPolicy.unmuted);
+  final follow =               button(to:   UserTopicVisibilityPolicy.followed);
+  final unfollow =             button(from: UserTopicVisibilityPolicy.followed,
+                                      to:   UserTopicVisibilityPolicy.none);
+
+  final store = PerAccountStoreWidget.of(context);
+  final channelMuted = store.subscriptions[channelId]?.isMuted;
+  final visibilityPolicy = store.topicVisibilityPolicy(channelId, topic);
+
+  // TODO(server-7): simplify this condition away
+  final supportsUnmutingTopics = store.connection.zulipFeatureLevel! >= 170;
+  // TODO(server-8): simplify this condition away
+  final supportsFollowingTopics = store.connection.zulipFeatureLevel! >= 219;
+
+  final optionButtons = <ActionSheetMenuItemButton>[];
+  if (channelMuted != null && !channelMuted) {
+    switch (visibilityPolicy) {
```

Comment:
nit:
```suggestion
  if (channelMuted != null && !channelMuted) {
    // Channel is subscribed and not muted.
    switch (visibilityPolicy) {
```

And then for the next case:
```dart
    // Channel is muted.
```

I had the thought that these conditionals were a bit opaque and required some unpacking by the reader. Then I looked at the handy linked zulip-mobile code and saw it has comments for exactly that need :slightly_smiling_face: — so let's carry those over.